sound/soc/sof/pm.c

```diff
@@ -119,8 +119,15 @@ static int sof_resume(struct device *dev, bool runtime_resume)
 	 * D0 substate.
 	 */
 	if (!runtime_resume && sof_ops(sdev)->set_power_state &&
-	    old_state == SOF_DSP_PM_D0)
+	    old_state == SOF_DSP_PM_D0) {
+		ret = snd_sof_dma_trace_trigger(sdev, SNDRV_PCM_TRIGGER_START);
+		if (ret < 0) {
+			dev_err(sdev->dev,
+				"error: snd_sof_dma_trace_trigger: start: %d\n", ret);
+		} else
+			sdev->dtrace_is_enabled = true;
```

Comment:
Same comment as below, the error print and modifying dtrace_is_enabled would be better handled in trace.c and just have a simple function call here.